contracts/libs/gov/gov-pool/GovPoolCreate.sol

```diff
@@ -245,6 +245,14 @@ library GovPoolCreate {
             bytes4 selector = actions[i].data.getSelector();
             uint256 executorSettings = govSettings.executorToSettings(actions[i].executor);
 
+            if (
+                executorSettings == uint256(IGovSettings.ExecutorType.INTERNAL) &&
+                actions[i].value != 0 &&
+                selector == IGovPool.delegateTreasury.selector
+            ) {
+                continue;
+            }
```

Comment:
```suggestion
  
                require(executorSettings == uint256(IGovSettings.ExecutorType.INTERNAL) &&
                selector == IGovPool.delegateTreasury.selector, "GovPool: invalid internal data");
```

And delete the selector from the next `require`

Reply:
It will revert on any selector other than delegateTreasury

Or you mean 
`if (msg.value > ) {your changes}`
?

Reply:
Yep, let's do that with `if` and also add one to the next block to only execute it if value is 0